src/django_htmx/middleware.py

```diff
@@ -112,3 +114,21 @@ def triggering_event(self) -> Any:
             except json.JSONDecodeError:
                 value = None
         return value
+
+
+def redirect_middleware(get_response):
+    """Middleware to handle redirects for htmx requests for login required views"""
+
+    def middleware(request):
+        response = get_response(request)
+        if request.headers.get("HX-Request") and response.status_code == 302:
+            location = urlparse(response["Location"])
+            if location.path == settings.LOGIN_URL:
```

Comment:
`@login_required` might use a different `login_url`, so this is fragile. Wrapping `login_required` would solve this